gtk/gtkfilechooserwidget.c

```diff
@@ -4042,6 +4045,45 @@ add_cwd_to_sidebar_if_needed (GtkFileChooserWidget *impl)
   g_object_unref (cwd_file);
 }
 
+static void
+browse_places_button_changed_cb (GtkFileChooserWidget *impl)
+{
+  gboolean active;
+  GValue gvalue = G_VALUE_INIT;
+  GtkFileChooserWidgetPrivate *priv = impl->priv;
+
+  active = gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (priv->browse_places_button));
+
+  /* First, make sure position-set is FALSE or the GtkRevealer won't animate */
+  g_value_init (&gvalue, G_TYPE_BOOLEAN);
+  g_value_set_boolean (&gvalue, 0);
+  g_object_set_property (G_OBJECT (priv->browse_widgets_hpaned), "position-set", &gvalue);
```

Comment:
you can use g_object_set (priv->browse_widgets_hpaned, "position-set", FALSE, NULL);